src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -25,36 +25,84 @@ impl {{ target_kind_name }} {
+                                            reader.read_{{ target_kind_name }}()?;
+                                        {% endif %}
         {% else %}
-            {% if data_model == "Struct"%}reader.step_in()?;{% endif %}
-            let mut data_model = {{ target_kind_name }}::default();
+            {% if abstract_data_type == "Struct"%}reader.step_in()?;{% endif %}
+            let mut abstract_data_type = {{ target_kind_name }}::default();
             while reader.next()? != StreamItem::Nothing {
                 if let Some(field_name) = reader.field_name()?.text() {
                     match field_name {
-
-                     {% for statement in statements -%}
-                            {{ statement }}
-                     {% endfor %}
+                        {% for field in fields -%}
+                            {% if field.value | is_built_in_type == false %}
+                                {% if abstract_data_type is object and abstract_data_type["Sequence"] | length != 0 %}
+                                 "{{ field.name }}" => { abstract_data_type.{{ field.name }} = {
+                                     let mut values = vec![];
+                                     reader.step_in()?;
+                                     while reader.next()? != StreamItem::Nothing {
+                                        {% if abstract_data_type["Sequence"] | is_built_in_type == false %}
+                                            values.push({{ abstract_data_type["Sequence"] }}::read_from(reader)?);
+                                        {% else %}
+                                            values.push(reader.read_{{ abstract_data_type["Sequence"] }}()?);
+                                        {% endif %}
+                                    }
+                                    values
+                                 };}
+                                 {% elif abstract_data_type == "Struct" %}
+                                 "{{ field.name }}" => { abstract_data_type.{{ field.name }} = {{ field.value }}::read_from(reader)?; }
+                                 {% endif %}
+                            {% else %}
+                            "{{ field.name }}" => { abstract_data_type.{{ field.name }} = reader.read_{{ field.value }}()?; }
+                            {% endif %}
+                        {% endfor %}
                      _ => {}
                     }
                 }
             }
-            {% if data_model == "Struct"%}reader.step_out()?;{% endif %}
+            {% if abstract_data_type == "Struct"%}reader.step_out()?;{% endif %}
         {% endif %}
-     Ok(data_model)
+     Ok(abstract_data_type)
     }
-    {% endif %}
 
     pub fn write_to<W: IonWriter>(&self, writer: &mut W) -> IonResult<()> {
-        {% for statement in write_statements -%}
-           {{ statement }}
-        {% endfor %}
+        {% if abstract_data_type == "Value" %}
```

Comment:
(🗺️ PR tour) All the code from `CodeGenerator::generate_write_api` has been moved to this template for ease of maintaining the code and a single place to store all the template related details.